src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -208,9 +216,21 @@ private void AddSecrets(TransactionBuildContext context)
                 return;
 
             Wallet wallet = this.walletManager.GetWalletByName(context.AccountReference.WalletName);
-
+            Key privateKey; 
             // get extended private key
-            var privateKey = Key.Parse(wallet.EncryptedSeed, context.WalletPassword, wallet.Network);
+            string cacheKey = Encoding.Default.GetString(Hashes.Hash256(Encoding.UTF8.GetBytes(wallet.EncryptedSeed)).ToBytes());
```

Comment:
@kogot I still don't understand why is the cackeKey calculated in this way. 😕

Reply:
@bokobza yeah reason was mostly to hide plain value cause we storing it in memory. If you so unhappy with that, I'm easy to use EncryptedSeed as key as is.

Reply:
I think it's ok. The encrypted seed is encrypted so no problem with that IMO.